AdyenComponents/UPI/UPIComponent.swift

```diff
@@ -0,0 +1,207 @@
+//
+// Copyright (c) 2022 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+@_spi(AdyenInternal) import Adyen
+import UIKit
+
+/// A component that provides a upi flows for UPI component.
+public final class UPIComponent: PaymentComponent,
+    PresentableComponent,
+    PaymentAware,
+    LoadingComponent {
+
+    private enum ViewIdentifier {
+        static let instructionsItem = "instructionsLabelItem"
+        static let upiFlowSelectionItem = "upiFlowSelectionSegmentedControlItem"
+        static let continueButtonItem = "continueButton"
+        static let generateQRCodeButtonItem = "generateQRCodeButton"
+        static let generateQRCodeContainerItem = "generateQRCodeLabelContainerItem"
+        static let virtualPaymentAddressInputItem = "virtualPaymentAddressInputItem"
+        static let qrCodeGenerationImageItem = "qrCodeGenerationImageItem"
+    }
+
+    private enum Constants {
+        static let upiCollect = "upi_collect"
+        static let upiQRCode = "upi_qr"
+        static let virtualPaymentAddress = "Virtual Payment Address"
+    }
+
+    /// Configuration for UPI Component.
+    public typealias Configuration = BasicComponentConfiguration
+
+    /// The context object for this component.
+    @_spi(AdyenInternal)
+    public var context: AdyenContext
+
+    public var paymentMethod: PaymentMethod { upiPaymentMethod }
```

Comment:
Let's add comments on each of the public properties

Reply:
Fixed. Here is the commit: 03dc881a644b5209e889ccefa98851afad4d1bcb

Reply:
@erenbesel This is not necessary if property of method are inherited.
Docs will use protocol's original comment